pkg/workflows/builder.go

```diff
@@ -1,35 +1,184 @@
 package workflows
 
-// The real implementations will come in a follow-up PR.
-// these stubs allow the code from the generators to compile.
-// A holistic view can be seen at https://github.com/smartcontractkit/chainlink-common/pull/695
+import (
+	"fmt"
+	"strconv"
+	"strings"
 
-type WorkflowSpecFactory struct{}
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+)
+
+// 1. Cap defines JSON schema for inputs and outputs of a cap.
+// Trigger: triggerOutputType := workflowBuilder.addTrigger(DataStreamsTrigger.ModifiedConfig{})
+// Adds metadata to the builder. Returns output type.
+// 2. Consensus: consensusOutputType := workflowBuilder.addConsensus(ConsensusConfig{
+// 	Inputs: triggerOutputType,
+// })
+
+type WorkflowSpecFactory struct {
+	spec           *WorkflowSpec
```

Comment:
How about just `SpecFactory` and even `Spec` since we are already in `package workflows` and callers do `workflows.WorkflowSpecFactory` & `workflows.WorkflowSpec`.
```suggestion
type SpecFactory struct {
	spec           *Spec
```

Reply:
Let's do that one in another PR since WorkflowSpec existed before any of the builder work.  In a follow up I can rename both together.